### PR TITLE
use rwmutex replace mutex and use RLock when get

### PR DESCRIPTION
**Pull request by rokkiter**
**Please provide a description of this PR:**

use rwmutex replace mutex and use RLock when get

### CODE REVIEW

File: pilot/pkg/model/cluster_local.go [hunk 1]
```diff
@@ -64,14 +64,14 @@ func NewClusterLocalProvider(e *Environment) ClusterLocalProvider {
 var _ ClusterLocalProvider = &clusterLocalProvider{}
 
 type clusterLocalProvider struct {
-	mutex sync.Mutex
+	mutex sync.RWMutex
```
Comment (howardjohn) [inline pilot/pkg/model/cluster_local.go:67]: could just use atomic.Pointer here?

**Review by howardjohn — APPROVED** [review]

**Review by hzxuzhonghu — APPROVED** [review]

File: pilot/pkg/model/cluster_local.go (hunk 1 above)
Comment (rokkiter) [reply pilot/pkg/model/cluster_local.go:67]: It should be possible to use atomic.Pointer here, but I prefer to merge this pr first, and then I submit a pr to handle all syncs that can be replaced by atomic.Pointer

Comment (hzxuzhonghu) [reply pilot/pkg/model/cluster_local.go:67]: for most cases it is recommended to use mutex

File: cni/pkg/ambient/server.go [hunk 2]
```diff
@@ -43,7 +43,7 @@ type Server struct {
 	namespaces kclient.Client[*corev1.Namespace]
 	pods       kclient.Client[*corev1.Pod]
 
-	mu         sync.Mutex
+	mu         sync.RWMutex
```
Comment (hzxuzhonghu) [inline cni/pkg/ambient/server.go:46]: unreleated to this pr: i think we donot need to store ztunnel pod here

**Pull request merged**